ci-operator/step-registry/set-sample-operator/disconnected/set-sample-operator-disconnected-commands.sh

```diff
@@ -22,38 +22,80 @@ function config_samples_operator()
+        echo "CA is ready, skip config..."
+        return 0
+    fi
+
+    # get the QE additional CA
+    QE_ADDITIONAL_CA_FILE="/var/run/vault/mirror-registry/additional_ca"
+    REGISTRY_HOST=`echo ${MIRROR_PROXY_REGISTRY} | cut -d \: -f 1`
+    # Configuring additional trust stores for image registry access, details: https://docs.openshift.com/container-platform/4.11/registry/configuring-registry-operator.html#images-configuration-cas_configuring-registry-operator
+    run_command "oc create configmap registry-config --from-file=\"${REGISTRY_HOST}..5000\"=${QE_ADDITIONAL_CA_FILE} --from-file=\"${REGISTRY_HOST}..6001\"=${QE_ADDITIONAL_CA_FILE} --from-file=\"${REGISTRY_HOST}..6002\"=${QE_ADDITIONAL_CA_FILE}  -n openshift-config"; ret=$?
+    if [[ $ret -eq 0 ]]; then
+        echo "set the proxy registry ConfigMap successfully."
+    else
+        echo "!!! fail to set the proxy registry ConfigMap"
+        run_command "oc get configmap registry-config -n openshift-config -o yaml"
+        return 1
+    fi
+    run_command "oc patch image.config.openshift.io/cluster --patch '{\"spec\":{\"additionalTrustedCA\":{\"name\":\"registry-config\"}}}' --type=merge"; ret=$?
     if [[ $ret -eq 0 ]]; then
-        echo "Mirror tag images to the Proxy registry successfully."
+        echo "set additionalTrustedCA successfully."
     else
-        echo "!!! Fail to mirror tag images to the Proxy registry"
-        run_command "cat /tmp/.dockerconfigjson"
+        echo "!!! Fail to set additionalTrustedCA"
+        run_command "oc get image.config.openshift.io/cluster -o yaml"
         return 1
     fi
 }
 
+function mirror_tag_images () {
+    echo "quay.io/openshift-qe-optional-operators/ocp4-index:latest=MIRROR_REGISTRY_PLACEHOLDER/openshift-qe-optional-operators/ocp4-index:latest
+registry.redhat.io/ubi8/ruby-30:latest=MIRROR_REGISTRY_PLACEHOLDER/ubi8/ruby-30:latest
+registry.redhat.io/ubi8/ruby-27:latest=MIRROR_REGISTRY_PLACEHOLDER/ubi8/ruby-27:latest
+registry.redhat.io/ubi7/ruby-27:latest=MIRROR_REGISTRY_PLACEHOLDER/ubi7/ruby-27:latest
+registry.redhat.io/rhscl/ruby-25-rhel7:latest=MIRROR_REGISTRY_PLACEHOLDER/rhscl/ruby-25-rhel7:latest
+registry.redhat.io/rhscl/mysql-80-rhel7:latest=MIRROR_REGISTRY_PLACEHOLDER/rhscl/mysql-80-rhel7:latest
+registry.redhat.io/rhel8/mysql-80:latest=MIRROR_REGISTRY_PLACEHOLDER/rhel8/mysql-80:latest
+registry.redhat.io/rhel8/httpd-24:latest=MIRROR_REGISTRY_PLACEHOLDER/rhel8/httpd-24:latest
+" > /tmp/tag_images_list
+
+    sed -i "s/MIRROR_REGISTRY_PLACEHOLDER/${MIRROR_PROXY_REGISTRY}/g" "/tmp/tag_images_list"
+    run_command "cat /tmp/tag_images_list"
+    run_command "cat /tmp/new-dockerconfigjson"
+    run_command "oc image mirror -f \"/tmp/tag_images_list\"  --insecure=true -a \"/tmp/new-dockerconfigjson\" --skip-missing=true --skip-verification=true --keep-manifest-list=true --filter-by-os='.*'"
+
+}
+
 run_command "oc get config.samples.operator.openshift.io cluster"; ret=$?
```

Comment:
I think we also need `set_proxy` function before running most of oc command against clusters, Note that, but we do not need proxy when mirror images.

Reply:
Good catch! Will have a try.